benches/cbor.rs

```diff
@@ -0,0 +1,221 @@
+                            0
+                        } else if info < 28 {
+                            2usize.pow(info as u32 - 24)
+                        } else {
+                            return Err(Error::empty() /*custom(span, format!("Invalid argument: {}", info))*/)
+                        };
+                        Ok(cfg.exactly(num))
+                    })
+            )
+                .map_slice(int_out);
+
+            let uint = read_int.map(CborZero::Int);
+            let nint = read_int.map(|i| CborZero::Int(-1 - i));
+            // TODO: Handle indefinite lengths
+            let bstr = read_int
+                .then_with_ctx(
+                    any()
+                        .repeated()
+                        .configure(|cfg, ctx| {
+                            cfg.exactly(*ctx as usize)
+                        })
+                        .map_slice(CborZero::Bytes)
+                );
+
+            let str = read_int
+                .then_with_ctx(
+                    any()
+                        .repeated()
+                        .configure(|cfg, ctx| {
+                            cfg.exactly(*ctx as usize)
+                        })
+                        .map_slice(|slice| CborZero::String(std::str::from_utf8(slice).unwrap()))
+                );
+
+            let array = read_int
+                .then_with_ctx(
+                    data
+                        .clone()
+                        .with_ctx(())
+                        .repeated()
+                        .configure(|cfg, ctx| {
+                            cfg.exactly(*ctx as usize)
+                        })
+                        .collect::<Vec<_>>()
+                        .map(CborZero::Array)
+                );
+
+            let map = read_int
+                .then_with_ctx(
+                    data.clone()
+                        .then(data.clone())
+                        .with_ctx(())
+                        .repeated()
+                        .configure(|cfg, ctx| {
+                            cfg.exactly(*ctx as usize)
+                        })
+                        .collect::<Vec<_>>()
+                        .map(CborZero::Map)
+                );
+
+            let simple = |num: u8| any()
+                .try_map(move |n, _| if n & 0b1_1111 == num {
+                    Ok(())
+                } else {
+                    Err(Error::empty() /*custom(span, format!("Invalid simple identifier {}", n))*/)
+                });
+
+            let float_simple = choice((
+                simple(20).to(CborZero::Bool(false)),
+                simple(21).to(CborZero::Bool(true)),
+                simple(22).to(CborZero::Null),
+                simple(23).to(CborZero::Undef),
+                simple(26).ignore_then(any()
+                    .repeated_exactly::<4>()
+                    .collect::<_, _, [_; 4]>()
+                    .map(f32::from_be_bytes)
```

Comment:
Out of interest, is this type hint necessary? It would be unfortunate if it was given the call to `map` just below.

Reply:
They're not, they were before the map was added, but now it appears rustc has enough info to figure it out